src/AddressRegistry.Api.Legacy/AddressMatch/Matching/AddressMatchBuilder.cs

```diff
@@ -77,66 +86,128 @@ public AddressMatchBuilder(AddressMatchQueryComponents query)
+        public void AddRrAddresses(IEnumerable<AddressDetailItem> rrAddresses)
+            => _rrAddresses.AddRange(rrAddresses);
 
-        public IEnumerable<StreetNameWrapper> AllStreetNames() =>
-            this.SelectMany(municipalityWrapper => municipalityWrapper).Distinct(StreetNameWrapper.Comparer);
+        public IEnumerable<StreetNameWrapper> AllStreetNames()
+            => this.SelectMany(municipalityWrapper => municipalityWrapper).Distinct(StreetNameWrapper.Comparer);
 
-        public IEnumerable<AddressDetailItem> AllAddresses() =>
-            AllStreetNames().SelectMany(s => s).Union(_rrAddresses).Distinct(StreetNameWrapper.AddressComparer);
+        public IEnumerable<AddressDetailItem> AllAddresses()
+            => AllStreetNames()
+                .SelectMany(s => s)
+                .Union(_rrAddresses)
+                .Distinct(StreetNameWrapper.AddressComparer);
 
         public IEnumerable<string> GetMatchRepresentationsForScoring()
         {
-            IEnumerable<string> municipalityNames = _municipalities.Values.Select(g => g.Name).Concat(new[] { Query.MunicipalityName }).Where(x => x != null).Distinct(StringComparer.InvariantCultureIgnoreCase);
+            var municipalityNames = _municipalities
+                .Values
+                .Select(g => g.Name)
+                .Concat(new[] { Query.MunicipalityName })
+                .Where(x => x != null)
+                .Distinct(StringComparer.InvariantCultureIgnoreCase);
 
-            IEnumerable<StreetNameWrapper> relevantStreetNames = AllStreetNames();
-            //if addresses found, forget about the streetnames without addresses
+            var relevantStreetNames = AllStreetNames();
+
+            // if addresses are found, forget about the streetnames without addresses
             if (AllAddresses().Any())
                 relevantStreetNames = relevantStreetNames.Where(s => s.Any());
 
-            IEnumerable<string> streetNames = relevantStreetNames.Select(sn => sn.StreetName.NameDutch).Concat(new[] { Query.StreetName }).Where(x => x != null).Distinct(StringComparer.InvariantCultureIgnoreCase);
-
-            Func<string, string, string, string> createRepresentation = (municipalityName, streetName, houseNumber) =>
-                string.Format("{0}{1}{2}{3},{4}{5}", streetName,
-                    !string.IsNullOrEmpty(houseNumber) ? string.Concat(" ", houseNumber) : string.Empty,
-                    !string.IsNullOrEmpty(Query.Index) ? string.Concat(" bus ", Query.Index) : string.Empty,
-                    !string.IsNullOrEmpty(Query.BoxNumber) ? string.Concat(" bus ", Query.BoxNumber) : string.Empty,
-                    !string.IsNullOrEmpty(Query.PostalCode) ? string.Concat(" ", Query.PostalCode) : string.Empty,
+            // TODO: Use GetDefaultName() instead of NameDutch?
+            var streetNames = relevantStreetNames
+                .Select(sn => sn.StreetName.NameDutch)
```

Comment:
Should we use `GetDefaultName()` instead of `NameDutch`?

Reply:
More a business question. We could look through all names, but for now it was limited to Dutch.
So unless we look through all, I suggest we don't change it. 

There is already change request to look through all on the backlog.

Reply:
Ok